src/Control/Monad/Aff/Bus.purs

```diff
@@ -50,20 +55,38 @@ type BusRW = Bus (read ∷ Cap, write ∷ Cap)
 
 -- | Creates a new bidirectional Bus which can be read from and written to.
 make ∷ ∀ eff a. Aff (avar ∷ AVAR | eff) (BusRW a)
-make = Bus <$> makeEmptyVar
+make = do
+  cell ← makeEmptyVar
+  consumers ← makeVar mempty
+  let
+    loop = do
+      attempt (takeVar cell) >>= traverse_ \res → do
+        vars ← takeVar consumers
+        putVar mempty consumers
+        sequence_ (foldl (\xs a → putVar res a : xs) mempty vars)
+        loop
+  _ ← forkAff loop
```

Comment:
This might need to be an unsafe fork (`liftEff <<< launchAff`) unless you want to let users to control the lifecycle of the loop with `supervised`.

Reply:
Changing that line to `_ ← liftEff (launchAff (forkAff loop))` explodes the tests with:

```
/home/creek/work/purescript-aff-bus/output/Control.Monad.Aff/foreign.js:513
                throw util.fromLeft(step);
                ^

TypeError: bhead is not a function
    at run (/home/creek/work/purescript-aff-bus/output/Control.Monad.Aff/foreign.js:274:20)
    at Object.run (/home/creek/work/purescript-aff-bus/output/Control.Monad.Aff/foreign.js:623:13)
    at __do (/home/creek/work/purescript-aff-bus/output/Control.Monad.Aff/index.js:91:11)
    at runSync (/home/creek/work/purescript-aff-bus/output/Control.Monad.Aff/foreign.js:90:20)
    at run (/home/creek/work/purescript-aff-bus/output/Control.Monad.Aff/foreign.js:319:22)
    at /home/creek/work/purescript-aff-bus/output/Control.Monad.Aff/foreign.js:620:15
    at drain (/home/creek/work/purescript-aff-bus/output/Control.Monad.Aff/foreign.js:120:9)
    at Object.enqueue (/home/creek/work/purescript-aff-bus/output/Control.Monad.Aff/foreign.js:141:11)
    at Object.run (/home/creek/work/purescript-aff-bus/output/Control.Monad.Aff/foreign.js:619:23)
    at __do (/home/creek/work/purescript-aff-bus/output/Control.Monad.Aff/index.js:91:11)
* ERROR: Subcommand terminated with exit code 1
```

Looks like it's maybe a bug in Aff?

Reply:
https://github.com/slamdata/purescript-aff/pull/145